gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-oidc/src/main/java/io/gravitee/am/gateway/handler/oauth2/service/response/IDTokenResponse.java

```diff
@@ -47,6 +45,7 @@ public void setIdToken(String idToken) {
     @Override
     public String buildRedirectUri() {
         UriBuilder uriBuilder = UriBuilder.fromURIString(getRedirectUri());
+        // idToken required fragment
```

Comment:
Are we sure about this ? 

The comment about mention this : ```unless a different Response Mode was specified.```

Reply:
According to https://openid.net/specs/oauth-v2-multiple-response-types-1_0.html#ResponseModes
For id_token response type:

* `The default Response Mode for this Response Type is the fragment encoding and the query encoding MUST NOT be used` 

My interpretation is only fragment is allowed or another mode like jwt. wdyt ?


`3.  ID Token Response Type
This section registers a new Response Type, the id_token, in accordance with the stipulations in the OAuth 2.0 specification, Section 8.4. The intended purpose of the id_token is that it MUST provide an assertion of the identity of the Resource Owner as understood by the Authorization Server. The assertion MUST specify a targeted audience, e.g. the requesting Client. However, the specific semantics of the assertion and how it can be validated are not specified in this document.

id_token
When supplied as the response_type parameter in an OAuth 2.0 Authorization Request, a successful response MUST include the parameter id_token. The Authorization Server SHOULD NOT return an OAuth 2.0 Authorization Code, Access Token, or Access Token Type in a successful response to the grant request. If a redirect_uri is supplied, the User Agent SHOULD be redirected there after granting or denying access. The request MAY include a state parameter, and if so, the Authorization Server MUST echo its value as a response parameter when issuing either a successful response or an error response. The default Response Mode for this Response Type is the fragment encoding and the query encoding MUST NOT be used. Both successful and error responses SHOULD be returned using the supplied Response Mode, or if none is supplied, using the default Response Mode.
Returning the id_token in a fragment reduces the likelihood that the id_token leaks during transport and mitigates the associated risks to the privacy of the user (Resource Owner).`

Reply:
```Both successful and error responses SHOULD be returned using the supplied Response Mode, or if none is supplied, using the default Response Mode.```

Not sure how to read this ... but I still think we should rely on the supplied response mode.

Reply:
If token or id_token are part of the response_type, then the response_mode has to be fragment (or fragment.jwt) for security reason as spotted by https://openid.net/specs/oauth-v2-multiple-response-types-1_0.html#Security.

I will amend the PR.